### PR TITLE
Platform tests

### PR DESCRIPTION
**Pull request by verdverm**
Uses Python Behave to test various workflows:
- [x] workspace lifecycle
- [x] deployment lifecycle

To run:

```
# Python Setup
virtualenv -p python3 venv
source venv/bin/activate
pip install -r requirements.txt

# Environment Setup
export ASTRO_EMAIL="<email>" ASTRO_PASS="<password>" ASTRO_DOMAIN="<basedomain>"

# Run tests
cd platform
behave
```

or, with docker:

```
# Build image
docker built -t astronomer.io/platform-tests .

# Environment Setup
export ASTRO_EMAIL="<email>" ASTRO_PASS="<password>" ASTRO_DOMAIN="<basedomain>"

# Run tests
docker run --rm -it \
  -e ASTRO_USER="$ASTRO_USER" \
  -e ASTRO_PASS="$ASTRO_PASS" \
  -e ASTRO_DOMAIN="$ASTRO_DOMAIN" \
  astronomer.io/platform-tests
```

### CODE REVIEW

**Comment by verdverm** [comment]
@sjmiller609 can you give this a try? Instructions are in the PR description now.

I got the following error when trying to create a deployment:

```
{'errors': [{'message': "Cannot read property 'defaultExtraAu' of undefined", 'locations': [{'line': 3, 'column': 5}], 'path': ['createDeployment'], 'extensions': {'code': 'INTERNAL_SERVER_ERROR'}, 'level': 'ERROR', 'timestamp': '2020-02-20T18:44:31'}], 'data': {'createDeployment': None}}
```

I'm attempting to imitate the way astro-cli does this

- CLI Query: https://github.com/astronomer/astro-cli/blob/master/deployment/deployment.go#L23
- CLI GraphQL: https://github.com/astronomer/astro-cli/blob/master/houston/queries.go#L18
- Test Query/GraphQL: https://github.com/astronomer/astronomer_e2e_test/blob/feature-platform-tests/platform/sdk/deployments.py#L26

@schnie any ideas about this?

**Comment by sjmiller609** [comment]
Hey Tony I'll be happy to take a look tomorrow. I'm going to head out early at 3 today, I won't have time to properly review.

**Comment by verdverm** [comment]
@sjmiller609 Ready for final review

Docker Hub Image: https://hub.docker.com/r/astronomerinc/ap-e2e-test/tags

**Comment by sjmiller609** [comment]
so I will just merge this whenever you want, but I was thinking it may make sense to merge into CI first since you are not blocked to push the image, then if any changes are required we can clean up in here then merge, but whatever order you want is fine with me.

**Comment by verdverm** [comment]
@sjmiller609 

I think we wait to merge this until we have it working in CI, for the reasons you mentioned.

File: platform/api/deployments.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
+import json
+
+
+def list(client, workspace):
```
Comment (sjmiller609) [inline platform/api/deployments.py]: hey I was copy/pasting this for a script I'm writing and it made me notice that "list" is a build-in function to python so you might consider re-naming.

File: platform/api/workspaces.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
+import json
+
+
+def list(client):
```
Comment (sjmiller609) [inline platform/api/workspaces.py]: same idea here.

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+                    tag_and_push $tag
+                  else
+                    echo "This docker tag already exists. Skipping the Docker push!"
+                  fi
+              else
+                tag_and_push $tag
+              fi
+            done
+  get-tag:
+    description: "Set an environment variable to this release branch with an incremented patch version"
```
[the remark this answers is not in the document]
Comment (verdverm) [reply .circleci/config.yml]: I saw that, meant to ask you about updating the message or if we want to actually bump. Sounds like we should update the message?

File: platform/features/002_deployments_cli.feature [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+      When (API) we create a workspace "deploymentTests" for "testing various deployments"
+      Then a new workspace is returned
+
+  Scenario: Switch to workspace via CLI
+    Given we are authenticated
+    And   we have a workspace
+      When (CLI) we can switch to workspace "deploymentTests"
+      Then a new workspace is returned
+
+  Scenario: FIX ME, RE-Authenticate with CLI after workspace switch
```
[the remark this answers is not in the document]
Comment (verdverm) [reply platform/features/002_deployments_cli.feature]: Yeah, there is a scenario that looks to be a bug, not sure if it's in the CLI or the platform. This test captures that this is happening, and handles the work around so we can continue testing downstream features.

File: platform/features/steps/deployments-cli.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+@when('(CLI) we create a "{executor}" deployment "{name}" for "{description}"')
+def step_impl(context, executor, name, description):
+    output = deployments.create(context.client, name, description, executor)
+    assert output is not None
+    print("OUTPUT", output)
+
+    lines = output.splitlines()
+    print("LINES-PRE", lines)
+    # Trim header
+    line = lines[1]
```
[the remark this answers is not in the document]
Comment (verdverm) [reply platform/features/steps/deployments-cli.py]: no, we are looking for a specific, singular line

File: platform/api/deployments.py (hunk 1 above)
Comment (verdverm) [reply platform/api/deployments.py]: good point, I'll update to `listing`

File: platform/features/steps/auth.py [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+import cli
+
+@given('we have credentials')
+def step_impl(context):
+    context.email = os.getenv("ASTRO_USER")
+    context.password = os.getenv("ASTRO_PASS")
+    assert context.email is not None
+    assert context.password is not None
+
+    context.baseDomain = os.getenv("ASTRO_DOMAIN", "staging.astronomer.io")
```
[the remark this answers is not in the document]
Comment (verdverm) [reply platform/features/steps/auth.py]: done

**Pull request merged**